src/app/reducers/manuscript.reducer.ts

```diff
@@ -287,13 +301,35 @@ function handleLinkAffiliations(state: ManuscriptHistory, action: Action<LinkAff
     }
   });
 
+  newManuscript.affiliations = getReorderedAffiliations(newManuscript.authors, newManuscript.affiliations);
+
   return {
     past: [...state.past, newDiff],
     present: newManuscript,
     future: []
   };
 }
 
+function getReorderedAffiliations(authors: Person[], affiliations: Affiliation[]): Affiliation[] {
+  const newAffiliations = affiliations.map((affiliation) => ({ ...affiliation, label: '' }));
+  let labelIndex = 1;
+  authors.forEach((author) => {
+    author.affiliations.forEach((affId) => {
+      const authorAffiliation = newAffiliations.find(({ id }) => id === affId);
+      if (authorAffiliation && !authorAffiliation.label) {
+        authorAffiliation.label = String(labelIndex++);
+      }
+    });
+  });
+  newAffiliations.forEach((affiliation) => {
+    if (!affiliation.label) {
+      affiliation.label = String(labelIndex++);
```

Comment:
Might need to double check with Nick what the label should be for an un-linked affilitation, was a '?' in Texture. Not sure if the same is required here.